text/0000-dockerfiles.md

```diff
@@ -0,0 +1,218 @@
+# Meta
+[meta]: #meta
+- Name: Support Dockerfiles
+- Start Date: 2021-06-30
+- Author(s): sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: [RFC0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md), [RFC#167](https://github.com/buildpacks/rfcs/pull/167)
+- Depends on: [RFC#172](https://github.com/buildpacks/rfcs/pull/172)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces functionality for customizing base images, as an alternative to stackpacks ([RFC0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md)).
+
+# Motivation
+[motivation]: #motivation
+
+[RFC0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md) introduces complexity by defining an API that allows buildpacks to modify base images. To avoid this complexity, we could rely on generated Dockerfiles for base image manipulation. This would simplify the original proposal by, e.g., only requiring a copy of the buildpack on the build-time base image.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes that we replace stackpacks with dynamically-generated build-time and runtime Dockerfiles that act as pre-build hooks.
+These hooks would participate in detection and execute before the buildpack build process.
+
+For a given application, a build that uses hooks could be optimized by creating a more narrowly-scoped builder that does not contain hooks.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Note: kaniko, buildah, BuildKit, or the original Docker daemon may be used to apply Dockerfiles at the platform's discretion. 
```

Comment:
And what about Podman, is it counted?

Reply:
it works reasonably well for me on Linxu, however the VM podman in macOS and Windows not so much

Reply:
IIUC, Podman uses buildah under the hood for building containers and provides only a subset of buildah's functionalities so if the buildah is mentioned, we should be fine.

Reply:
Intent is not to be restrictive. This is just clarifying that existing build tools that support the Dockerfile format may be used to implement this RFC.

Reply:
is this meant to say a platform like VMWare/Heroku/Google would _provide_ the thing that applies Dockerfiles..or is this saying that `lifecycle` is able to choose any of these tools?

Reply:
```suggestion
Note: kaniko, buildah, BuildKit, or the original Docker daemon may be used to apply Dockerfiles at the platform's discretion. The order of operations would be something like the following:
* analyze
* detect
* restore
* <new lifecycle phase> run hooks' bin/build, output Dockerfiles are written to a volume
* <new lifecycle phase OR platform> apply Dockerfiles to run image (could run in parallel with below two)
* <new lifecycle phase OR platform> apply Dockerfiles to build image
* build
* export
```

Reply:
Now that I'm thinking about it, we could run bin/build for hooks during `detect`, since hooks don't use the cache...